unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarRenderer.cs

```diff
@@ -495,6 +431,106 @@ private IEnumerator LoadAvatar()
             }
         }
 
+        private IEnumerator WearAnyMissingClothes(IEnumerable<WearableItem> wearables,
+            Action<Exception> error,
+            Action<List<WearableItem>> completed)
+        {
+            var wearedAndReplacedCategories = wearables.SelectMany(wearable =>
+                {
+                    var replacedByWearable = wearable.GetReplacesList(model.bodyShape) ?? new string[0];
+                    return new[] {wearable.data.category}.Concat(replacedByWearable);
+                })
+                .Distinct();
+            var missingClothesReplacements = new [] {Categories.LOWER_BODY, Categories.UPPER_BODY}
+                .Except(wearedAndReplacedCategories)
+                .Select(RequestWearableReplacement);
```

Comment:
`RequestWearableReplacement` is not a naive function as it sends a wearable request. We are 100% positive that `Select` works as a `ForEach` here and will not send a request twice for a given wearable? I'm very wary about the intensive usage of LINQ on this code as it makes the logic not very easy to follow.

Reply:
Replaced most of the LINQ operations, i think it got a little bit simpler to understand. It should not happen to send the request twice due using `Select`, since later, at `LoadWearables`, is yielded where is checks if its already resolved. The promise instance is the same